harness/nans.js

```diff
@@ -7,6 +7,17 @@ description: |
     weak basis for assertions regarding the consistent canonicalization of NaN
     values in Array buffers.
 ---*/
-var distinctNaNs = [
-  0/0, Infinity/Infinity, -(0/0), Math.pow(-1, 0.5), -Math.pow(-1, 0.5)
+
+var NaNs = [
+  () => NaN,
+  () => Number.NaN,
+  () => NaN * 0,
+  () => 0/0,
+  () => Infinity/Infinity,
+  () => -(0/0),
+  () => Math.pow(-1, 0.5),
+  () => -Math.pow(-1, 0.5),
+  () => Number("Not-a-Number"),
 ];
+
+var distinctNaNs = NaNs.map(n => n());
```

Comment:
In my experimenting with https://npmjs.com/get-nans , i saw a number of engines where something like this would cause it to get canonicalized, such that inlining it directly into the array was the only way to retain the ability to distinguish them.

Reply:
Ah, good to know, I will adjust accordingly.